Sources/PrimerSDK/Classes/User Interface/TokenizationViewModels/FormsTokenizationViewModel/CardFormPaymentMethodTokenizationViewModel.swift

```diff
@@ -368,20 +368,22 @@ class CardFormPaymentMethodTokenizationViewModel: PaymentMethodTokenizationViewM
         
         return Promise { seal in
             firstly {
-                self.validateReturningPromise()
+                return self.validateReturningPromise()
             }
             .then { () -> Promise<Void> in
                 return self.presentCardFormViewController()
             }
             .then { () -> Promise<Void> in
                 return self.awaitUserInput()
             }
+            .then { () -> Promise<Void> in
+                return self.updateButtonUI()
+            }
             .then { () -> Promise<Void> in
                 self.didStartTokenization?()
                 return self.dispatchActions()
             }
             .then { () -> Promise<Void> in
-                self.updateButtonUI()
```

Comment:
Can you double check we don't need this to update the button UI after surcharge being applied?